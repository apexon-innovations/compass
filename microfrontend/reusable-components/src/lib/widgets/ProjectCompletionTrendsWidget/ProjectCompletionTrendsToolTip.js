import React from 'react'
import PropTypes from 'prop-types'

const ProjectCompletionTrendsToolTip = props => {
  const { active, payload, label } = props

  if (active && props.lineIndex) {
    const data = payload[props.lineIndex] ? payload[props.lineIndex] : false
    if (data) {
      const showData = data.payload[`${data.dataKey}Data`]
      return (
        <div className="customToolTip sprintPoint text-center">
          <div className="header">
            <div className="sprintName">{label}</div>
          </div>
          <div className="content">
            <ul>
              <li>
                <span className="label">{showData.label}</span>
                <span className="value">{showData.completed || '0'}</span>
              </li>
              <li>
                <span className="label">Squads</span>
                <span className="value">{showData.squads || '0'}</span>
              </li>
            </ul>
          </div>
        </div>
      )
    }
    return null
  }
  return null
}

ProjectCompletionTrendsToolTip.propTypes = {
  active: PropTypes.bool,
  payload: PropTypes.any,
  label: PropTypes.any,
  lineIndex: PropTypes.any,
}

export default ProjectCompletionTrendsToolTip
