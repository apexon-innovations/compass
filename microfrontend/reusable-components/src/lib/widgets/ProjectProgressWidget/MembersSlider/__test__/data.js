const data = {
  data: [
    {
      email: 'Pradeep.shrivastava@infostretch.com',
      name: 'Pradeep Shrivastava',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933',
    },
    {
      email: 'vijay.solanki@infostretch.com',
      name: 'Vijay Solanki',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621',
    },
    {
      email: 'suraj.bhosale@infostretch.com',
      name: 'Suraj Bhosale',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950',
    },
    {
      email: 'yatin.bhavsar@infostretch.com',
      name: 'Yatin Bhavsar',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973',
    },
    {
      email: 'vishal.kumar@infostretch.com',
      name: 'Vishal kumar',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930',
    },
    {
      email: 'gunjan.shah@infostretch.com',
      name: 'Gunjan Shah',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
    },
    {
      email: 'namra.porecha@infostretch.com',
      name: 'Namra Porecha',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375',
    },
    {
      email: 'purushottam.Khandebharad@infostretch.com',
      name: 'Purushottam Khandebharad',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437',
    },
    {
      email: 'amit.ghadge@infostretch.com',
      name: 'Amit Ghadge',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269',
    },
  ],
}

export default data
