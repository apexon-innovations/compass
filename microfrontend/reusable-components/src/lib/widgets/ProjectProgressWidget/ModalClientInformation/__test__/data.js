const data = {
  data: {
    id: '5ee31002d48d514e801cb10b',
    name: 'Compass',
    clientName: 'Infostretch',
    initials: 'C',
    category: 'All (Dev & QA & Support)',
    iconLocation:
      'http://sampleicon.com/image.png',
    billingType: 'Internal',
    startDate: 'Wed Jan 01 00:00:00 UTC 2020',
    endDate: 'Thu Jan 02 00:00:00 UTC 2020',
    resources: {
      actual: 10,
      sow: 10,
    },
    healthMetrics: {
      overall: 'R',
      periodicReview: 'G',
      scope: 'A',
      deliverables: 'A',
      customer: 'G',
      schedule: 'R',
      financials: 'G',
      people: 'G',
      compliance: 'G',
      startUp: 'G',
      process: 'G',
      planning: 'G',
    },
    projectManager: [],
    deliveryManager: [
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
        nestProjects: ['Infostretch - DES'],
      },
    ],
    accountManager: [
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
        nestProjects: ['Infostretch - DES'],
      },
    ],
    teamMembers: [
      {
        email: 'Pradeep.shrivastava@infostretch.com',
        name: 'Pradeep Shrivastava',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933',
      },
      {
        email: 'vijay.solanki@infostretch.com',
        name: 'Vijay Solanki',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621',
      },
      {
        email: 'suraj.bhosale@infostretch.com',
        name: 'Suraj Bhosale',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950',
      },
      {
        email: 'yatin.bhavsar@infostretch.com',
        name: 'Yatin Bhavsar',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973',
      },
      {
        email: 'vishal.kumar@infostretch.com',
        name: 'Vishal kumar',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930',
      },
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
      },
      {
        email: 'namra.porecha@infostretch.com',
        name: 'Namra Porecha',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375',
      },
      {
        email: 'purushottam.Khandebharad@infostretch.com',
        name: 'Purushottam Khandebharad',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437',
      },
      {
        email: 'amit.ghadge@infostretch.com',
        name: 'Amit Ghadge',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269',
      },
    ],
  },
  fileUploaderUrl:
    'https://compass-dev-api.infostretch.com/psr-service/project/5ee31002d48d514e801cb10b/logo',
}

export default data
