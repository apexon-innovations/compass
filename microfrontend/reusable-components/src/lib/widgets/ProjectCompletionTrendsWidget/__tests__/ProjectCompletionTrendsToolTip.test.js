/* eslint-disable */
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { sleep, projectId, baseUrl } from '../../../../__tests__/commonTestFunction'
import ProjectCompletionTrendsToolTip from '../ProjectCompletionTrendsToolTip'
import { toolTipData, toolTipData1 } from './data'

const props = { ...toolTipData }

describe(`ProjectCompletionTrendsToolTip`, () => {
  let container = null

  beforeEach(() => {
    // set up a DOM element as a render target
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container)
    container.remove()
    container = null
  })

  it(`Snapshot`, async () => {
    act(() => {
      render(
        <Router>
          <ProjectCompletionTrendsToolTip {...props} />
        </Router>,
        container,
      )
    })
    await act(() => sleep(500))
    expect(container).toMatchSnapshot()
  })

  it(`Snapshot No Data`, async () => {
    act(() => {
      render(
        <Router>
          <ProjectCompletionTrendsToolTip />
        </Router>,
        container,
      )
    })
    await act(() => sleep(500))
    expect(container).toMatchSnapshot()
  })

  it(`Snapshot No Line Index`, async () => {
    act(() => {
      render(
        <Router>
          <ProjectCompletionTrendsToolTip {...toolTipData1} />
        </Router>,
        container,
      )
    })
    await act(() => sleep(500))
    expect(container).toMatchSnapshot()
  })
})
