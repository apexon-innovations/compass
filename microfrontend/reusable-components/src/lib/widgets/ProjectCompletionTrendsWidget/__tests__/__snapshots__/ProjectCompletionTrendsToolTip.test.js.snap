// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCompletionTrendsToolTip Snapshot 1`] = `
<div>
  <div
    class="customToolTip sprintPoint text-center"
  >
    <div
      class="header"
    >
      <div
        class="sprintName"
      >
        Prev Sprint 4
      </div>
    </div>
    <div
      class="content"
    >
      <ul>
        <li>
          <span
            class="label"
          >
            Story Points
          </span>
          <span
            class="value"
          >
            253
          </span>
        </li>
        <li>
          <span
            class="label"
          >
            Squads
          </span>
          <span
            class="value"
          >
            4
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ProjectCompletionTrendsToolTip Snapshot No Data 1`] = `<div />`;

exports[`ProjectCompletionTrendsToolTip Snapshot No Line Index 1`] = `<div />`;
